lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -93,6 +95,7 @@ defmodule LambdaEthereumConsensus.ForkChoice do
         prune_old_states(last_finalized_checkpoint.epoch, new_finalized_checkpoint.epoch)
 
         GenServer.cast(from, {:block_processed, block_root, true})
+        persist_fork_choice_store(new_store)
```

Comment:
I would persist before calling the cast. As a general rule it's useful that the state is stored and consistent before others are made aware of it. As a matter of fact, if we persist it before launching the recompute_head async task, we can remove the persist call in recompute-head.

Reply:
That's true. Done!